wgsl/index.bs

```diff
@@ -8285,43 +8350,97 @@ Unless explicitly permitted below, an attribute [=shader-creation error|must not
     Note: This attribute is used as a notational convention to describe which
     built-in functions can be used in [=const-expressions=].
 
-  <tr><td><dfn noexport dfn-for="attribute">`diagnostic`</dfn>
+</table>
+
+## `diagnostic` ## {#diagnostic-attr}
+
+<pre class=include>
+path: syntax/diagnostic_attr.syntax.bs.include
+</pre>
+<pre class=include>
+path: syntax/diagnostic_control.syntax.bs.include
+</pre>
+
+<table class='data builtin'>
+  <caption><dfn noexport dfn-for="attribute">`diagnostic`</dfn> Attribute</caption>
+  <tr>
+    <td style="width:10%">Valid Values
     <td>Two parameters.
 
         The first parameter is a [=syntax/severity_control_name=].
 
         The second parameter is a [=syntax/diagnostic_rule_name=] token
         specifying a [=diagnostic/triggering rule=].
-
+  <tr>
+    <td>Description
     <td>Specifies a [=range diagnostic filter=].  See [[#diagnostics]].
 
         More than one [=attribute/diagnostic=] attribute may be specified on a syntactic form,
         but they [=shader-creation error|must=] specify different [=diagnostic/triggering rules=].
 
-  <tr><td><dfn noexport dfn-for="attribute">`group`</dfn>
+</table>
+
+## `group` ## {#group-attr}
+
+<pre class=include>
+path: syntax/group_attr.syntax.bs.include
+</pre>
+
+<table class='data builtin'>
+  <caption><dfn noexport dfn-for="attribute">`group`</dfn> Attribute</caption>
+  <tr>
+    <td style="width:10%">Valid Values
     <td>[=shader-creation error|Must=] be a [=const-expression=] that [=type rules|resolves=] to an [=i32=] or [=u32=].<br>
         [=shader-creation error|Must=] be non-negative.
+  <tr>
+    <td>Description
     <td>[=shader-creation error|Must=] only be applied to a [=resource=] variable.
 
     Specifies the binding group of the resource.
     See [[#resource-interface]].
 
-  <tr><td><dfn noexport dfn-for="attribute">`id`</dfn>
+</table>
+
+## `id` ## {#id-attr}
+
+<pre class=include>
+path: syntax/id_attr.syntax.bs.include
+</pre>
+
+<table class='data builtin'>
+  <caption><dfn noexport dfn-for="attribute">`id`</dfn> Attribute</caption>
+  <tr>
+    <td style="width:10%">Valid Values
     <td>[=shader-creation error|Must=] be a [=const-expression=] that [=type rules|resolves=] to an [=i32=] or [=u32=].<br>
         [=shader-creation error|Must=] be non-negative.
+  <tr>
+    <td>Description
     <td>[=shader-creation error|Must=] only be applied to an [=override-declaration=] of [=scalar=] type.
 
     Specifies a numeric identifier as an alternate name for a
     [=pipeline-overridable=] constant.
 
-  <tr><td><dfn noexport dfn-for="attribute">`interpolate`</dfn>
-    <td>One or two parameters.
+</table>
+
+## `interpolate` ## {#interpolate-attr}
+
+<pre class=include>
+path: syntax/interpolate_attr.syntax.bs.include
```

Comment:
I have a preference to match the diagnostic attribute of splitting the parameters into a separate rule, but consistency one way or the other is more preferable to me.

Reply:
I am neutral for this. If @dneto0 also thinks it is better to rule them than make it short and part of syntax, I can certainly go ahead and update like that.

Reply:
Considering how diagnostic control is part of both attribute and directive, I believe it can have its treatment in this case and does not significantly influence consistency. Otherwise, taking the parameters of all attributes into rules creates a vertically severe bloat. In favor of progress, I am marking it as resolved, but if you, sir, @alan-baker, would like to provide guidance on the criteria of splits for which attributes to take out arguments as a rule without denormalizing diagnostic_control (the diagnostic parameters that go to both attribute and directive), do please mark this as unresolved, and we can ping @dneto0 feedback. I prefer keeping diagnostic_control as diagnostic_control and making the rest of the attributes list the parameters, as they are right now in rules but with their section. Thank you!